### PR TITLE
chore: Adding amount field to telemetry request

### PR DESCRIPTION
**Pull request by goergisn**
## Summary
- Adding amount value to telemetry requests

### CODE REVIEW

File: Adyen/Analytics/AnalyticsProvider/TelemetryTracker.swift [hunk 1]
```diff
@@ -30,24 +30,28 @@ public enum TelemetryFlavor {
 
 @_spi(AdyenInternal)
 public protocol TelemetryTrackerProtocol {
-    func sendTelemetryEvent(flavor: TelemetryFlavor)
+    func sendTelemetryEvent(flavor: TelemetryFlavor, amount: Amount?)
```
Comment (descorp) [inline Adyen/Analytics/AnalyticsProvider/TelemetryTracker.swift]: how about simply use `context` on init ?
It is bit odd - only `AdyenContext` uses `AnalyticsProviderProtocol`  but it need `amount` from `AdyenContext`..

File: Adyen/Core/Core Protocols/PresentableComponent.swift [hunk 2]
```diff
@@ -69,8 +69,16 @@ public protocol TrackableComponent: Component {
...
-        context.analyticsProvider.sendTelemetryEvent(flavor: flavor)
+        context.analyticsProvider.sendTelemetryEvent(flavor: flavor, amount: amount)
+    }
+}
+
+@_spi(AdyenInternal)
+extension TrackableComponent where Self: PaymentAware & PaymentMethodAware {
+    
+    public func sendTelemetryEvent() {
+        sendTelemetryEvent(withAmount: payment?.amount)
```
Comment (erenbesel) [inline Adyen/Core/Core Protocols/PresentableComponent.swift]: If you added the `PaymentAware` check here, then I think you don't need to re-conform to the function on all the below places

File: Adyen/Analytics/AnalyticsProvider/TelemetryTracker.swift (hunk 1 above)
Comment (goergisn) [reply Adyen/Analytics/AnalyticsProvider/TelemetryTracker.swift]: @descorp Did not think about this, thanks 🙏 
I added an `amount` requirement next to `checkoutAttemptId` in the AnalyticsProvider protocol which resulted in way less changes. Let me know what you think.

File: Adyen/Core/AdyenContext/AdyenContext.swift [hunk 3]
```diff
@@ -43,10 +49,18 @@ public final class AdyenContext: PaymentAware {
...
     }
 
     @_spi(AdyenInternal)
     public func update(payment: Payment?) {
         self.payment = payment
     }
 }
+
+private extension AdyenContext {
+    func updateAnalyticsProvider() {
```
Comment (erenbesel) [inline Adyen/Core/AdyenContext/AdyenContext.swift]: Nitpick but maybe a name like `updateAnalyticsAmount` etc?

File: Adyen/Analytics/AnalyticsProvider/AnalyticsProvider.swift [hunk 4]
```diff
@@ -27,8 +27,10 @@ public struct AnalyticsConfiguration {
 @_spi(AdyenInternal)
 public protocol AnalyticsProviderProtocol: TelemetryTrackerProtocol {
     
-    var checkoutAttemptId: String? { get }
+    /// The payment-amount to be tracked
+    var amount: Amount? { get set }
```
Comment (descorp) [inline Adyen/Analytics/AnalyticsProvider/AnalyticsProvider.swift]: do we need "set" ?

Comment (goergisn) [reply Adyen/Analytics/AnalyticsProvider/AnalyticsProvider.swift]: https://github.com/Adyen/adyen-ios/pull/1287/files#diff-3a0d5cb797232c8b37393f98b8fa3b114fb17f74fa72d82b016231ee5bf39448R64

This line nees it as the `AdyenContext` receives `any AnalyticsProviderProtocol`

Comment (descorp) [reply Adyen/Analytics/AnalyticsProvider/AnalyticsProvider.swift]: fair, it should not be possible to change amount on Context, but for now it is the only way 🤔

**Review by descorp — DISMISSED** [review]

File: Adyen/Analytics/AnalyticsProvider/AnalyticsProvider.swift [hunk 5]
```diff
@@ -24,12 +24,20 @@ public struct AnalyticsConfiguration {
     public init() { /* Empty implementation */ }
 }
 
+@_spi(AdyenInternal)
+/// Additional fields to be provided with a ``TelemetryRequest``
+public struct AdditionalAnalyticsFields {
```
Comment (descorp) [inline Adyen/Analytics/AnalyticsProvider/AnalyticsProvider.swift:29]: Nice :)

**Review by erenbesel — DISMISSED** [review]

**Review by erenbesel — DISMISSED** [review]

**Review by erenbesel — APPROVED** [review]

**Review by neelSharma12 — APPROVED** [review]

**Pull request merged**